cpp/src/arrow/dataset/test_util.h

```diff
@@ -589,6 +589,23 @@ class FileFormatScanMixin : public FileFormatFixtureMixin<FormatHelper>,
     }
     ASSERT_EQ(row_count, GetParam().expected_rows());
   }
+  // Ensure batch_size is respected
+  void TestScanBatchSize() {
+    auto reader = GetRecordBatchReader(schema({field("f64", float64())}));
+    auto source = this->GetFileSource(reader.get());
+
+    this->SetSchema(reader->schema()->fields());
+    auto fragment = this->MakeFragment(*source);
+
+    int64_t row_count = 0;
+    opts_->batch_size = 16;
```

Comment:
I think all of our test cases have # rows evenly divisible by 16.  Maybe make this 15 or something odd so that we ensure we test cases where the read doesn't split into even batches.

Reply:
Good point, I bumped it to 17 which is also prime!